SConstruct

```diff
@@ -1074,207 +1108,212 @@ env['python_cmd_esc'] = quoted(env['python_cmd'])
+    env['python_package'] = 'full'  # Allow 'y' as a synonym for 'full'
+elif env['python_package'] in ['none', 'n']:
+    if env['python2_package'] == 'default':
+        env['python2_package'] = 'n'
+    if env['python3_package'] == 'default':
+        env['python3_package'] = 'n'
+
+for v in [2, 3]:
+    if env['python{}_package'.format(v)] == 'y':
+        env['python{}_package'.format(v)] = 'full'  # Allow 'y' as a synonym for 'full'
+    elif env['python{}_package'.format(v)] == 'n':
+        env['python{}_package'.format(v)] = 'none'  # Allow 'n' as a synonym for 'none'
+
+require_python = any([env[p] == 'full' for p in ['python_package', 'python2_package', 'python3_package']])
+want_python = any([env[p] == 'default' for p in ['python_package', 'python2_package', 'python3_package']])
+
+if require_python or want_python:
     # Check for Cython:
     try:
         import Cython
-    except ImportError:
-        warn_no_python = True
-    else:
         cython_version = LooseVersion(Cython.__version__)
-        if cython_version < cython_min_version:
-            python_message += ("Cython not found or incompatible version: "
-                               "Found {0} but {1} or newer is required.\n".format(cython_version,
-                                                                                  cython_min_version))
-            warn_no_python = True
-        else:
-            print('INFO: Using Cython version {0}.'.format(cython_version))
-
-    # Test to see if we can import the specified array module
-    script = '\n'.join(("from __future__ import print_function",
-                        "from distutils.sysconfig import *",
-                        "print(get_python_version())",
-                        "try:",
-                        "    import numpy",
-                        "    print(numpy.__version__)",
-                        "except ImportError as err:",
-                        "    import sys",
-                        "    print(err, file=sys.stderr)",
-                        "    print('0.0.0')",
-                        "import site",
-                        "try:",
-                        "    print(site.getusersitepackages())",
-                        "except AttributeError:",
-                        "    print(site.USER_SITE)"))
-
-    if env['python_array_home']:
-        script = "sys.path.append({})\n".format(env['python_array_home']) + script
+        assert cython_version >= cython_min_version
+    except (ImportError, AssertionError):
+        message = ("Cython not found or incompatible version: "
+                   "Found {0} but {1} or newer is required.".format(cython_version,
+                                                                    cython_min_version))
+        if require_python:
+            print('ERROR: ' + message)
+            sys.exit(1)
+        elif want_python:
+            print('WARNING: ' + message)
+            env['python_package'] = 'minimal'
```

Comment:
Does setting `python_package` at this point have the desired effect, or does the code above which sets `python2_package` and `python3_package` based on `python_package` need to come after this point?

Reply:
I think this has the desired effect (at least, I'm pretty sure its doing what I expect it to do; whether that's desired or not is a different question 😄). 

If any of `python_package`, `python2_package`, or `python3_package` are `full` (after the code above, `y` and `new` become `full`), `require_python` is `True`, and if Cython can't be imported with the `python_cmd` interpreter, we want to error out, because the build can't be completed. (It would probably be better to check that Cython is able to be imported by the same Python that is running SCons, since `python_cmd`, if specified, will point to the Python we want to install for, not necessarily the one we want to build with. However, this PR doesn't change the behavior of the check for Cython.)

If all of `python_package`, `python2_package`, and `python3_package` are `'default'` (or any combination are `'none'` (but not all of them `'none'`) with any one of them as `'default'`), then `require_python` will be `False` and `want_python` will be `True`. In that case, if we can't find Cython, `python_package` will be set to `'minimal'` and the flow will continue to check what version of Python is specified in `python_cmd`. Then, the Python in `python_cmd` will be used to build the minimal interface for that version of Python (provided that the `python*_package` that matches the version of Python in `python_cmd` is not specified).

That's what I wanted this line to do, and I'm pretty sure that's what its doing.